src/main/java/com/sda/games/checkers/model/Game.java

```diff
@@ -1,71 +1,155 @@
+        Player blackPlayer = new Player(2, Player.blackPlayerName = scanner.nextLine(), false);
 
-    public Game(Player[] players) {
-        this.players = players;
+
+        initialize(whitePlayer, blackPlayer);
+        getBoard().printBoard();
     }
 
     public void initialize(Player player1, Player player2) {
-        players[0] = player1;
-        players[1] = player2;
+        this.players = new ArrayList<>();
+        players.add(player1);
+        players.add(player2);
 
+        this.board = new Board();
         board.resetBoard();
 
-        if (player1.getWhite()) {
-            this.currentTurn = player1;
+        if (player1.isWhite()) {
+            this.currentTurnPlayer = player1;
         } else {
-            this.currentTurn = player2;
+            this.currentTurnPlayer = player2;
         }
 
+        this.movesPlayed = new ArrayList<>();
         movesPlayed.clear();
     }
 
-    public boolean isEnd()
-    {
-        return this.getStatus() != GameStatus.ACTIVE;
-    }
+    public boolean makeMove() throws Exception {
 
-    public GameStatus getStatus() {
-        return status;
-    }
+        Player currentPlayer = getCurrentTurnPlayer();
 
-    public void setStatus(GameStatus status) {
-        this.status = status;
-    }
+        System.out.println(getCurrentTurnPlayer().getName() + " move.");
+
+        Pattern movePattern = Pattern.compile("[a-hA-H][1-8]");
+
+        String startSpotXY = "";
+        char charStartY;
+        char charStartX;
+        int startX;
+        int startY;
+        String endSpotXY = "";
+        char charEndX;
+        char charEndY;
+        int endX;
+        int endY;
+        Piece sourcePiece;
+
+        // Choosing checker to move
+        while (true) {
+            while (true) {
+                System.out.println("Which checker to move?");
+                startSpotXY = scanner.nextLine();
+
+                Matcher matcher = movePattern.matcher(startSpotXY);
+                boolean isInputValid = matcher.matches();
+
+                charStartX = startSpotXY.charAt(0);
+                charStartY = startSpotXY.charAt(1);
+                startX = charStartX - 97;
+                startY = charStartY - 49;
+
+                // Start piece validation
+                if (!isInputValid) {
+                    System.out.println("Invalid input!");
+                } else {
+                    if (board.getBoardSpot(startX, startY) == null) {
+                        System.out.println("No checker here!");
+                    } else {
+                        sourcePiece = board.getBoardSpot(startX, startY).getPiece();
+                        if (sourcePiece.isStartPieceValid(board, currentTurnPlayer, startX, startY)) {
+                            break;
+                        }
+                    }
+                }
+            }
+            break;
+        }
+        // Choosing where to move
+        while (true) {
```

Comment:
masz 2 prawie takie same fragmenty kodu, to powinna byc funkcja wywolana 2 razy :)